WooCommerce/Classes/ViewRelated/Products/Categories/ProductCategoryListViewController.swift

```diff
@@ -99,23 +135,32 @@ private extension ProductCategoryListViewController {
         viewModel.observeReloadNeeded { [weak self] in
             self?.tableView.reloadData()
         }
-        viewModel.observeCategoryListStateChanges { [weak self] syncState in
-            guard let self = self else { return }
-            switch syncState {
-            case .initialized:
-                break
-            case .syncing:
-                if self.viewModel.categoryViewModels.isEmpty {
-                    self.displayGhostContent()
+
+        viewModel.$syncCategoriesState.combineLatest(viewModel.$categoryViewModels)
+            .sink { [weak self] syncState, models in
+                guard let self = self else { return }
+                self.emptyStateViewController.view.isHidden = true
+                self.tableView.isHidden = false
+                switch syncState {
+                case .initialized:
+                    break
+                case .syncing:
+                    if self.viewModel.categoryViewModels.isEmpty {
```

Comment:
Super nit: Can we use `models.isEmpty` just to maintain consistency with line no 157?

Reply:
Good point! Updated in b183235e596dd240012740a6f4e3d60206ee8da8.